source/slang/slang-lower-to-ir.cpp

```diff
@@ -1056,6 +1056,11 @@ static void addLinkageDecoration(
     {
         builder->addExternCppDecoration(inst, mangledName);
     }
+    if (as<InterfaceDecl>(decl->parentDecl) &&
+        decl->parentDecl->hasModifier<ComInterfaceAttribute>())
+    {
+        builder->addExternCppDecoration(inst, decl->getName()->text.getUnownedSlice());
+    }
```

Comment:
I am not sure I understand this logic. It looks like this is saying that any IR inst we create for a child of a `[COM] interface` is being given the IR-level `__extern_cpp` decoration, and that is using the un-mangled original declaration name of the member.

AFAICT, the main effect that would have is to ensure that the emitted C++ declarations for a COM interface use the same member function names as the original, but that doesn't seem to be required for binary compatibility. If we emit C++ that calls a method `foo_0` or `_S3foo` instead of `foo`, will it actually break anything?

Reply:
Right, names shouldn't matter for binary compatibility. But if we want to generate header files from a COM interface exported by Slang, we probably need to use unmangled names?

Reply:
I continue to believe that generation of C++ headers based on Slang code will be best approached with something AST-based rather than IR-based.